### DISCUSSION TITLE
#30DaysOfMSGraph (✅ Completed)

### DISCUSSION THREAD

**Discussion by japhletnwamu** [microsoft/30daysof#85] [Announcements]
### **#30DaysOfMicrosoftGraph Discussion Forum**

**Welcome to this Discussion Forum**
This discussion forum was created for you. Here are a couple of things we want you to do here:

- Introduce yourself (Tell us who you are, what you do, where you are joining us from, and what you hope to benefit from this program)
- Ask any questions you have. We've got moderators willing to support you
- Update us on your learning progress, we love to hear it

We hope you all enjoy this program.

**Comment by japhletnwamu** [comment]
I'm Japhlet Nwamu a Cloud Advocate at Microsoft based in Nigeria. I'm part of the Next Generation Experience team and focus on Microsoft 365, Power Platform and Artificial Intelligence.

Hope to empower you with the right tools, resources and support you need to succeed and grow exponentially.

**Reply by solehuddinnorman** [reply]
Hi Japhlet, make sure that Someleze introduce himself here too like he promised :). You and your team are very energetic and helpful.

Jokes aside, I am very new to this, developing and programming. I hope to learn something here that I can use at work, trying to utilise Graph with Power Automate, and have fun in the process!

**Reply by japhletnwamu** [reply]
Sure, you would have fun with us @solehuddinnorman. And yes, I'll get @dikodev to do his introduction. 😁😁

**Comment by aycabas** [comment]
I'm Ayça Baş, Senior Cloud Developer Advocate at Microsoft based in Dubai, I'm a part of Microsoft 365 Advocacy team. I love working with Microsoft Graph, Teams, Functions, Cognitive Services, Event Hubs, Logic Apps and building inspiring solutions to help developers achieve more with Microsoft Cloud! 💜

Looking forward to meeting everyone joining 30 Days of Microsoft Graph Challenge and I'll be here to help throughout the month of November to  support you during your journey with Microsoft Graph!🦒

**Reply by TIDIALLO** [reply]
Thank you @aycabas

**Comment by waldekmastykarz** [comment]
👋 hey everyone, I'm Waldek, and I'm a Developer Advocate for Microsoft 365 based out of the Netherlands. You can ask me anything about building apps on Microsoft 365! I'm looking forward to meeting you all and hack away with you!

**Comment by RabebOthmani** [comment]
Hello, I'm Rabeb! I'm a senior product manager with the Microsoft Graph team. I'm so looking forward to meeting everyone and assist whenever possible with the challenge. Feel free to reach out to me with any questions :)

**Comment by jasonjoh** [comment]
👋 I'm Jason, a content developer with the Microsoft Graph team. I'm the person behind our [build your first app tutorials](https://learn.microsoft.com/graph/tutorials) and [quick starts](https://developer.microsoft.com/graph/quick-start). Please feel free to reach out to me with any questions!

**Comment by sebastienlevert** [comment]
👋 Hello everybody! I'm Seb and I'm a Senior Product Manager for our Microsoft Graph Toolkit and JavaScript SDKs!  Always feel free to ask anything about building apps with Microsoft Graph on the web, with NodeJS, or even if you have drums question 🥁! Looking forward to hacking with all of you!

**Comment by ktasolutions** [comment]
Hey everyone! I am Moses and I am a DevOps Engineer with Investec! I am looking forward to starting this challenge and learning a lot. I am running into an issue provisioning the prerequisites:
1. When adding the cell number it rejects it as invalid
2. Other prerequisites are activating GitHub Student Benefits or Azure Student Credit but I am not a student. Do I need to do this or is it optional 
Can someone help me or point me in the right direction to get help. Thank you

**Reply by aycabas** [reply]
Hi Moses, if you already have a GitHub account, you can use the existing one, GitHub for Students is optional. Do you get rejects for cell number when you are registering for the Microsoft 365 Developer Program?

**Reply by samueljaja00** [reply]
Getting a repo directory can we used a document to save in a cloud from Microsoft

**Comment by ChiamakaThelma** [comment]
Hi everyone!

I'm Chiamaka Thelma Okwara, a Planning & Controls Engineer with Frazimex Engineering in Nigeria, who is slowly transitioning into RPA.

I have registered for the Microsoft Graph Challenge. I have an existing office 365 developer account, and currently just got my first GitHub account.

Can't wait to explore what this challenge holds.

Thanks to the organizers for putting this together.

**Reply by waldekmastykarz** [reply]
Welcome! Awesome to have you with us! 👏

**Comment by balaupp** [comment]
I also just joined and excited to learn Graph!

**Comment by mattwalshdev** [comment]
Hi everyone, my names Matt and I just joined as I was looking for a LDAP connection library for M365 and then stumbled upon Graph and this course!

**Comment by solehuddinnorman** [comment]
Hi everyone, my name is Sol from Malaysia. I came across this programme when looking for a solution for a PA Flow for my work. Looking forward to learn from this!

Is this programme happening very often or this is one off kind of thing?

**Reply by japhletnwamu** [reply]
Hi Sol. We usually run 30DaysOf initiative for different Microsoft technologies/career paths (Power Platform, Data Analysis, Data Science) regularly. 30Days of Microsoft Graph is the first of this initiative focused on Microsoft Graph.

**Comment by NovoaDev** [comment]
Hello everyone, my name is Antonio from Spain. I am a Dynamics 365 - Business Central Developer. 

I am starting to learn Graph and I hope to learn a lot in this program <3.

Thank you very much for sharing knowledge.

**Comment by OfWhite** [comment]
Hello, I am Fabio,working as a Junior SW Engineer at beanTech Srl in Italy,really looking into getting deeper on the subject because I think it will enhance my CRM  knowledge and connect many tools that we usually work with.

**Comment by JonoSuave** [comment]
Hey I'm Jono! I'm a Developer of almost three years. At work I often get a chance to work on creating SharePoint applications. I'd like to level up in learning best practices of calling Microsoft Graph from Teams and SpFx applications.

**Comment by richardcarrigan** [comment]
Hey everyone! I'm Richard, a JS developer 👨‍💻, Power Platform expert, DevOps addict, and soccer fanatic ⚽.

Really excited to work through this challenge and connect with everyone. If anyone else wants to connect, I've posted my social links below. DMs are open 😉

Twitter: https://twitter.com/Rich_Carrigan
LinkedIn: https://www.linkedin.com/in/richardcarrigan/

**Comment by SRIDEVI24** [comment]
Hi,

Excited to join this challenge

**Comment by primado** [comment]
Hello everyone, I'm Prince Adimado. I'm a Computer Science student in Accra, Ghana, and a Beta Microsoft Learn Student Ambassador.  I heard about this program in Teams in the student ambassadors community. I've heard of Microsoft Graph before but I want to dedicate this month to learning the fundamentals to see where it takes me. 
Thank you

**Comment by aycabas** [comment]
## Welcome to the 30 Days Of Microsoft Graph!🦒

Hi everyone!🌸 

30 Days of Microsoft Graph Challenge started yesterday and we are so excited that you are interested in joining this journey with us! 🥳

Sharing the learning materials of the week below, you can find more information about the registration and preprequisites on our landing page: https://aka.ms/30DaysOfMSGraph 

## Week 1 - The Challenge begins!

* **Nov 01** - **3PM GMT Live Reactor Session ([Watch On Demand](https://www.youtube.com/watch?v=VkQbhmlSDuc)) :** Kickoff Meeting! Join the 30 Days of Microsoft Graph Challenge! 
* **Nov 02** - Work on the Microsoft Graph Challenge! [Learn Path: Microsoft Graph Fundamentals | Module: What is Graph?](https://docs.microsoft.com/en-us/training/modules/msgraph-intro-overview/) | Video instructions: [Microsoft Graph Fundamentals for Beginners](https://learn.microsoft.com/en-us/shows/beginners-series-to-microsoft-graph/)
* **Nov 03** - Work on the Microsoft Graph Challenge! [Learn Path: Microsoft Graph Fundamentals | Module: Configure a JavaScript application to retrieve Microsoft 365 data by using Microsoft Graph](https://docs.microsoft.com/en-us/training/modules/msgraph-javascript-app/) | Video instructions: [Microsoft Graph Fundamentals for Beginners](https://learn.microsoft.com/en-us/shows/beginners-series-to-microsoft-graph/)
* **Nov 04** - Work on the Microsoft Graph Challenge! [Learn Path: Microsoft Graph Fundamentals | Module: Access user photo information by using Microsoft Graph](https://docs.microsoft.com/en-us/training/modules/msgraph-user-photo-information/) | Video instructions: [Microsoft Graph Fundamentals for Beginners](https://learn.microsoft.com/en-us/shows/beginners-series-to-microsoft-graph/)


## Don't Forget!

All the learning materials are available to consume on your free time, this is a self-pace learning challenge. If you have questions or need help, please utilize this discussion channel, we are always here to help you 😊🚀 

If you haven't registered to the challenge yet, please register, so you can be eligible for the exciting prizes🏆: https://aka.ms/30DaysOfMSGraph/Register

**Comment by dikodev** [comment]
Hi Everyone! I'm Someleze Diko a Cloud Advocate at Microsoft based in South Africa. I'm part of the Next Generation Experience team and focus on Microsoft 365, Power Platform and Artificial Intelligence.

Hope to empower you with the right tools, resources and support you need to succeed and grow exponentially.

The 30DaysofMSGraph will be going on for the entirety of November so we're looking forward to be with you on this amazing journey!

PS: Yes, I'm late for the intro 😅

**Comment by NovoaDev** [comment]
# Error npx port auto(8080) different from redirect URI port.
In case someone gets the same thing that happens to me 😊. 
In my case **npx http-server** assign a different port than **8080** sometimes **8081**/**8082** depending on the ports it has busy.  This causes it to redirect to another service after getting the token and throwing an error.

## Error scenario
Service up on **8081**  
![DiferentPortNpx](https://github.com/NovoaDev/30-Days-Of-Microsoft-Graph-/blob/main/docs/res/portIssue/DiferentPortNpx.png)

Error when redirecting 
![Error8080](https://github.com/NovoaDev/30-Days-Of-Microsoft-Graph-/blob/main/docs/res/portIssue/Error8080.png)

## My solution
Manually assign a port that is free.
 1. Modify **package.json**. 
![PackageMod](https://github.com/NovoaDev/30-Days-Of-Microsoft-Graph-/blob/main/docs/res/portIssue/PackageMod.png)

 2. Modify redirection **aad.portal.azure** configuration.
![SpaaadappMod](https://github.com/NovoaDev/30-Days-Of-Microsoft-Graph-/blob/main/docs/res/portIssue/SpaaadappMod.png)

3. Modify **redirectUri** in archive auth.js.
![AuthjsMod](https://github.com/NovoaDev/30-Days-Of-Microsoft-Graph-/blob/main/docs/res/portIssue/AuthjsMod.png)

Raise the service again and it worked :P 

![200OK](https://github.com/NovoaDev/30-Days-Of-Microsoft-Graph-/blob/main/docs/res/portIssue/200OK.png)

**Reply by NovoaDev** [reply]
Idk why the images are not loading :[

**Reply by NovoaDev** [reply]
Is about this exercise:
https://learn.microsoft.com/en-us/shows/beginners-series-to-microsoft-graph/module-2-exercise-retrieve-an-access-token-by-using-msal

**Reply by japhletnwamu** [reply]
Hi @NovoaDev please submit this as an issue using this template: https://github.com/microsoft/30daysof/discussions/52.

**Comment by richardcarrigan** [comment]
# Unable to create new M365 dev subscription

I'm having trouble signing up for the M365 Developer Program. When I get to the phone number verification screen, I receive the following error message:

![image](https://user-images.githubusercontent.com/13327134/199624249-6ddc95d6-211d-4a5c-a888-06d43a939ca1.png)

I've had a developer M365 subscription before (expired on July 26, 2022, deleted on September 24, 2022). Would this prevent me from creating a new one?

**Reply by aycabas** [reply]
Hi @nihonjinboy85 thanks for letting us know. Let me check with the M365 Dev Program team and get back to you immediately.

**Reply by aycabas** [reply]
Hi @nihonjinboy85, you are right, the issue is caused by the existing expire subscription. Would you login to the existing account in M365 Developer Program and try to remove the expired subscription?

**Reply by richardcarrigan** [reply]
Thanks @aycabas . I'll give it a shot right away and let you know how it goes.

**Reply by richardcarrigan** [reply]
![Screenshot_20221103-082645.png](https://user-images.githubusercontent.com/13327134/199763302-d879493a-f43e-41e9-9d74-99e113b14d59.png)

And now we wait...

**Reply by richardcarrigan** [reply]
@aycabas , I just tried again. My profile seemed to have been successfully deleted, but once I signed back up, my previously deleted subscription shows up again, and there are no options to delete it (just 'Delete profile').

I'm really hoping this can get resolved soon, as I think I'll learn a lot from 30 Days Of MS Graph!

**Comment by rickyelias-avanade** [comment]
Hi all - My name is Ricky Elias and I'm joining the program from Brisbane, Australia.
I'm working as a consultant for Avanade - and focus on SharePoint, Search, CRM, Power Platform, Security and now MS Graph!
I'm passionate about using tech to make our lives better - and am guessing that graph will be a big help.
Hoping to be able to keep up the pace.

**Reply by pkbullock** [reply]
Hey @rickyelias-avanade, Im in Avanade UK :-), have fun building!

**Comment by aditi-gulati** [comment]
Hi Everyone👋, I am Aditi Gulati, present beta Microsoft Learn Student Ambassador from India. I am Passionate to work on Microsoft Technologies👩🏼‍💻 like Microsoft Graph, Windows, and Azure 💟. Really excited to work on this challenge, connect with everyone, and dedicate this month to learning MS Graph. 

LinkedIn: https://www.linkedin.com/in/aditi-gulati/

**Comment by isaacmtp** [comment]
are there any swags like tee's, stickers, bottles ... for those who participate

**Comment by karthickbharathwaj97** [comment]
Hello Everyone,
I am Karthick and I am a cloud enthusiast and a Front End developer joining this challenge from India. While I was trying to get the users Email via graph I have got this error
![graph error](https://user-images.githubusercontent.com/29711397/199804766-6b67e139-c449-48aa-b1a1-20ecfad69ce5.png)
I am continuing the documentation which was available here https://learn.microsoft.com/en-us/training/modules/msgraph-show-user-emails/6-exercise-show-users-email.
Could anyone please guide me through this error and tell what might be wrong?

**Reply by jasonjoh** [reply]
Hi @Depixeler - the error would indicate an issue with your tenant ID. Go to your app registration and double-check the value for **Directory (tenant) ID**. Make sure this GUID is the one that appears in the `authority` property in auth.js (per [this exercise](https://learn.microsoft.com/training/modules/msgraph-show-user-emails/4-exercise-connect-javascript-app)).

**Reply by karthickbharathwaj97** [reply]
I cross checked the Directory (tenant) ID. It is the same which appears to be in the code 
![image](https://user-images.githubusercontent.com/29711397/200038899-930d76ba-b452-40b3-ae8a-2a1ffc2035e7.png)
![image](https://user-images.githubusercontent.com/29711397/200039009-fd378c58-b37d-49df-abd2-2bca4b1f8c03.png)
Is there something wrong with my config?

**Reply by jasonjoh** [reply]
Thanks for verifying that. Is it possible that the account you are using with the app is not licensed for Exchange Online? Is the user you are logging into in the same Azure AD tenant that you registered your app in?

**Reply by karthickbharathwaj97** [reply]
Yes the account I am logging in is the same account which I used for register my app and get the Azure tenant Id. It is a normal outlook email I think I haven't registered for exchange online. Is there a way to check that out?

**Reply by jasonjoh** [reply]
We should verify what sort of account you're using. Is this an account in a work or school organization, or is it a personal account? If it's a personal account, have you purchased Microsoft 365 services, or is it a standard free account?

**Comment by frankyshoes** [comment]
Hi everyone! I'm Phillip, a Test Consultant from the UK. I've been studying towards various Microsoft Cloud exams this year and I'm looking forward to the 30 Days of Microsoft Graph Challenge. 😊

**Comment by mattwalshdev** [comment]
So this a question about the roadmap of the Graph API, specifically permissions. If I want my web application to say, look up who belongs to a group, I need to provide Application level permissions on _Group.Read.All_ (to list all the groups), then _Directory.Read.All_ (to get user info linked to their group member id).

Obviously I'm safely storing the secret keys to authenticate, but they have to be stored as environment variables _somewhere_, so while they are read only permissions, there is a lot of data that could be read from just those two permissions.

I love how when interfacing the API, if you don't have permissions, it still returns the object and just fills it with nulls, as opposed to 404s, so I wonder if there's a future where you could tick the actual object keys you want to view on each item, for example on _Directory.Read.All_, expand out and select only id, first name, last name, email address; thus reducing the potential data leak vector if the keys got into the wrong hands?

**Reply by aycabas** [reply]
Interesting suggestion Matt, @RabebOthmani do you have any comments on permissions roadmap?

**Comment by corriespondent** [comment]
Hi! I'm clicking through the various links to see what to expect and the one for week 3, take a break reminder app tutorial - https://aka.ms/TakeABreakReminder - is just going to GitHub's home page for me. Am I missing something?

**Reply by aycabas** [reply]
Hi @corriespondent good catch 😄 Take a break reminder workshop is just prepared and the link must be working now. That workshop is the learning material of week 3, so we'll cover it next week. But, if you are interested in taking a look, https://aka.ms/TakeABreakReminder is working now.

**Comment by StarsandPlanets99** [comment]


**Reply by aycabas** [reply]
Hi @StarsandPlanets99, can you please share the code snippet and screenshot of your browser debug. Let me help you find the issue.

**Reply by StarsandPlanets99** [reply]
Hi - thank you for your willingness. I started afresh today with a completey new install and app registration and resolved the issue.
Thanks

**Reply by aycabas** [reply]
Glad it's solved, did you find out what was causing the issue?

**Reply by StarsandPlanets99** [reply]
I think it was because I was trying to build on top of everything and so evolved  the app with the sign in, the profile photo, the emails, the calendar and then the files. The code copying in every exercise assumes you start from fresh everytime and so I think the code just got muddled. So user error really!

**Reply by aycabas** [reply]
Happens to me all the time, I can relate 🙂 I'm glad it's sorted 👏

**Comment by aycabas** [comment]
## This is week #2 🔥
Hi everyone!🌸

How is week 2 in the challenge going? Feel free to comment below 🥳

Sharing the learning materials of the week, you can find more information about the registration and preprequisites on our landing page: https://aka.ms/30DaysOfMSGraph

## Week 2 - Call the Microsoft Graph!
- **Nov 07** - 3PM GMT Live Reactor Session ([Watch On Demand](https://www.youtube.com/watch?v=Qx-80Snd2nw)): Build exciting apps with Microsoft Graph live!
- **Nov 08** - [Start calling the Microsoft Graph API in under 5 minutes! | Tips & Tricks](https://www.youtube.com/watch?v=f_3wc4UgqTI)
- **Nov 09** - Work on the Microsoft Graph Challenge! [Learn Path: Explore Microsoft Graph scenarios for JavaScript development | Module: Show a user's emails in a JavaScript app with Microsoft Graph](https://learn.microsoft.com/en-us/training/modules/msgraph-show-user-emails/)
- **Nov 10** - Work on the Microsoft Graph Challenge! [Learn Path: Explore Microsoft Graph scenarios for JavaScript development | Module: Access a user's calendar events in a JavaScript app with Microsoft Graph](https://learn.microsoft.com/en-us/training/modules/msgraph-access-user-events/)
- **Nov 11** - Work on the Microsoft Graph Challenge! [Learn Path: Explore Microsoft Graph scenarios for JavaScript development | Module: Download and upload files in a JavaScript app with Microsoft Graph](https://learn.microsoft.com/en-us/training/modules/msgraph-manage-files/)


## Don't Forget!
All the learning materials are available to consume on your free time, this is a self-pace learning challenge. If you have questions or need help, please utilize this discussion channel, we are always here to help you 😊🚀

If you haven't registered to the challenge yet, please register, so you can be eligible for the exciting prizes🏆: https://aka.ms/30DaysOfMSGraph/Register

**Reply by Naveennellikanti** [reply]
Hi Team,

Not sure if this is the right place to ask the query.

Hope you all are doing well,

I've a quick question here on "Nov 09 - Work on the Microsoft Graph Challenge" i could get details only based on email address, can you help if there is a way with SMTP & users id's? 

Thanks in Advance.

**Comment by chuff119** [comment]
My name is charles. im excited about learning about graph

**Reply by waldekmastykarz** [reply]
Great to have you with us, @chuff119!

**Comment by Avishasa** [comment]
Hey every one :) 
My name is Avishag and I'm a junior software developer.
I look forward to participate in this challenge, building an app for my team that we'll improve our collaboration, save time on administrative tasks, and allow us to focus our efforts in what we love - coding.

**Reply by waldekmastykarz** [reply]
Welcome! Awesome to have you with us, @Avishasa 👏

**Comment by aycabas** [comment]
# This is week 3!🔥

Hi everyone!🌸

How is week 3 in the challenge going? Feel free to comment below 🥳

Sharing the learning materials of the week, you can find more information about the registration and preprequisites on our landing page: https://aka.ms/30DaysOfMSGraph

## Week 3 - Connect the Dots
**Nov 14** - 3PM GMT Live Reactor Session ([Watch On Demand](https://www.youtube.com/watch?v=PK9S_GSmJA8)) : Build the "Take a break reminder" app with Microsoft Graph and Azure Functions
**Nov 15** - Practice: [Take a break reminder app tutorial](https://aka.ms/TakeABreakReminder)
**Nov 16** - Practice: [Build seamless automations to boost productivity with Microsoft Graph, Event Hubs and Functions](https://dev.to/azure/build-seamless-automations-to-boost-productivity-with-microsoft-graph-azure-event-hubs-and-functions-1ho8)
**Nov 17 - Submit your project proposal with Microsoft Graph: [Project Proposal Submission](https://aka.ms/30DaysOfMSGraph/Scenarios)**
**Nov 18** - 3PM GMT Live Reactor Session ([RSVP](https://aka.ms/30DaysOfMSGraph/Reactor/04) | [LIVE](https://www.youtube.com/watch?v=eRJvOdeBbBY)) : Ask the experts – Office hours to brainstorm capstone projects

## Don't Forget!
All the learning materials are available to consume on your free time, this is a self-pace learning challenge. If you have questions or need help, please utilize this discussion channel, we are always here to help you 😊🚀

If you haven't registered to the challenge yet, please register, so you can be eligible for the exciting prizes🏆: https://aka.ms/30DaysOfMSGraph/Register

### ℹ️ Don't forget to submit our project proposal this week: [Submit Here](https://aka.ms/30DaysOfMSGraph/Scenarios)

**Reply by StarsandPlanets99** [reply]
Thanks
Following the link for the Take a break reminder app leads me to two tutorials
Authentication App
Reminder App
Is the Authentication a pre-requisite to the Reminder App?
if not why not just link straight to the Reminder App.

Thanks

**Reply by StarsandPlanets99** [reply]
Also for this practice
https://dev.to/azure/build-seamless-automations-to-boost-productivity-with-microsoft-graph-azure-event-hubs-and-functions-1ho8
Step 1.6 asks to create a Consumer Group. This does not seem to be in the left menu - have Consumer Groups be replaced? If so what should the step be.

Thank you

**Reply by aycabas** [reply]
Hi @StarsandPlanets99, regarding your question about the Consumer Group, in your "Event Hubs Namespace", there is a tab called "Event Hub", you need to go to "Event Hub" tab and create a new Event Hub there. Then you need to select your Event Hub and you'll see Consumer Group. Please see below images:

![image](https://user-images.githubusercontent.com/36196437/201930403-05eda7ab-b02a-4eac-90c4-a9611a647fb9.png)

![image](https://user-images.githubusercontent.com/36196437/201930477-de418d3e-1e6d-4de9-b83f-567a7ab39638.png)

**Reply by glaucia86** [reply]
Hi, @StarsandPlanets99! Welcome to the 30 days of Microsoft Graph. My name is Glaucia Lemos and I am Developer Advocate here at Microsoft. 

No. It's not necessary to make the workshop 01. You can go forward from workshop 02. There's a kit getting started that I included in the session that you can download the application and start to focus just only in the Reminder logic.

Please any questions don't hesitate to ask.

Regards, 

Glaucia Lemos

**Reply by SRIDEVI24** [reply]
Hi @aycabas/ @waldekmastykarz 

I'm facing a issue in the  Practice: [Build seamless automations to boost productivity with Microsoft Graph, Event Hubs and Functions](https://dev.to/azure/build-seamless-automations-to-boost-productivity-with-microsoft-graph-azure-event-hubs-and-functions-1ho8)

![image](https://user-images.githubusercontent.com/67327691/203838022-89c726e6-a131-4e73-9091-05022f3fdf7e.png)

1.how to start the Microsoft Storage emulator? 
2.When 'func host start ' is given in the VSCode Terminal, it throws the below error :

PS D:\MS_GRAPH_CHALLENGE\EventHub_MS_Graph\SubscriptionFunction_timerTrigger> func
func : The term 'func' is not recognized as the name of a cmdlet, function, script file, or operable 
program. Check the spelling of the name, or if a path was included, verify that the path is correct and    
try again.
At line:1 char:1
+ func
+ ~~~~
    + CategoryInfo          : ObjectNotFound: (func:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

**Reply by SRIDEVI24** [reply]
> Hi @aycabas/ @waldekmastykarz
> 
> I'm facing a issue in the Practice: [Build seamless automations to boost productivity with Microsoft Graph, Event Hubs and Functions](https://dev.to/azure/build-seamless-automations-to-boost-productivity-with-microsoft-graph-azure-event-hubs-and-functions-1ho8)
> 
> ![image](https://user-images.githubusercontent.com/67327691/203838022-89c726e6-a131-4e73-9091-05022f3fdf7e.png)
> 
> 1.how to start the Microsoft Storage emulator? 2.When 'func host start ' is given in the VSCode Terminal, it throws the below error :
> 
> PS D:\MS_GRAPH_CHALLENGE\EventHub_MS_Graph\SubscriptionFunction_timerTrigger> func func : The term 'func' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. At line:1 char:1
> 
> * func
> * ```
>     + CategoryInfo          : ObjectNotFound: (func:String) [], CommandNotFoundException
>     + FullyQualifiedErrorId : CommandNotFoundException
>   ```

Hi @aycabas @waldekmastykarz ,

1.For the func error above, azure function cli has to be installed as    **npm i -g azure-functions-cli**  from the terminal
2.For the Microsoft Storage explorer ,  https://learn.microsoft.com/en-us/azure/storage/common/storage-use-emulator
    then installed azurite in Visual studio and started the azurite from visual studio,though the azurite service is running ( as azure functions require a storage account) but still i get the following error

![image](https://user-images.githubusercontent.com/67327691/204049508-a25f54ce-749a-49e6-a1b2-29f30c20f4d7.png)
![image](https://user-images.githubusercontent.com/67327691/204049653-d76cc5b4-1739-41eb-ab70-d9f509efb524.png)



How do I check if the subscription function for user is successfully run?..as it shows debugger listening..

**Comment by aborse555** [comment]
Hi! I am currently working on the "Taking a Break Reminder App" workshop and have been having issues related to compiling after running the command "npm run dev." I wanted to know if anyone had any solutions for this problem! Thank you for your help!

**Reply by waldekmastykarz** [reply]
Awesome to have you with us @aborse555! Could you please share some more information about the issue that you're experiencing that would help us troubleshoot it?

**Reply by aborse555** [reply]
Thank you @waldekmastykarz ! So after running the "npm run dev" command, the compiling of the client and server has been taking very long/does not finish. I have attached a screenshot of what comes up after running this command during section 5 of workshop 1.
![example](https://user-images.githubusercontent.com/116681133/202301906-05b6ee32-ea53-4086-8ebc-70c0e0a00cc4.JPG)

**Reply by waldekmastykarz** [reply]
Looking at the screenshot, there is the final event saying that compilation has completed, or am I missing something @aborse555?

**Reply by aborse555** [reply]
Yes, it works now! Thank you!

**Comment by aycabas** [comment]
## Did you pick your scenario for next week yet?🥳 

Feel free to choose one of the listed scenarios here: [Project Proposals](https://github.com/microsoft/30daysof/discussions/88#discussioncomment-4164387)

#### 🏆If you successfully complete your project, you will be interviewed on [Microsoft 365 Developer YouTube](https://aka.ms/m365youtube) and the interview video will be promoted across Microsoft socials 👏

#### 🏆 If your project gets selected for top 3, you will get a 1:1 mentorship session with Microsoft Graph and Advocacy Directors! 

Don't forget to submit your project proposal this week! 🥰 [Project Proposals](https://github.com/microsoft/30daysof/discussions/88#discussioncomment-4164387)

**Comment by balaupp** [comment]
I am working on the excercise in module 2, "Retrieve an access token". I have completed creating an app, grabbed the client and tenant ids correctly, and input them in the auth.js. I run 'npm start', everything compiles fine, opens the browser and displays the sig-in button. However, after clicking the "sign-in with microsoft" button, all i get is a blank screen. I have verified the "authority" and other contants are setup right. Could you point in the right direction as to what may be causing this? Thank you!

![image](https://user-images.githubusercontent.com/47118885/202536901-615bd441-3e5f-4788-ba53-d86d34f1d54c.png)

**Reply by aycabas** [reply]
Hi @balaupp, did you define `http://localhost:8080` as your redirect uri in the AAD app registration process? It might be causing an issue due to missing redirect. Let me know :)

**Reply by balaupp** [reply]
Yes, I did. 
![image](https://user-images.githubusercontent.com/47118885/202634237-ab1746b5-84dc-49bf-9f83-00a155b130e2.png)

**Reply by balaupp** [reply]
here in the code as well: 
![image](https://user-images.githubusercontent.com/47118885/202634357-52a406c2-98e8-4e0b-91ad-3224427540b1.png)

I have also checked my home network firsewalls (I used pi-hole etc) and it's all good as well. I am stumped.

**Reply by aycabas** [reply]
@waldekmastykarz do you have any opinion what could be the reason that auth pop up shows as blank?

**Reply by aycabas** [reply]
@balaupp did you set your aad app multi-tenant or single? Commenting out the `authority` in your auth code helps if you set your aad app as multi tenant.

**Reply by balaupp** [reply]
By Single, do you mean Single page app? Can you please point me to the place where I can verify this please? Thanks!

**Reply by balaupp** [reply]
Is this what you mean? : 
![image](https://user-images.githubusercontent.com/47118885/202635958-fd863b81-41b4-4841-98e7-cb72b264f29d.png)

**Reply by aycabas** [reply]
Yes! Seems like you set up your app as single tenant, means that your code should work fine as it is. @waldekmastykarz might have other ideas to help.

**Reply by balaupp** [reply]
@aycabas @waldekmastykarz When I debug, this is what I am seeing. Does it make sense to you? This is the code I pulled from the github and cloned it to a local folder. No changes other than to the Client ID and Tenant ID in auth.js.
![image](https://user-images.githubusercontent.com/47118885/202761638-f396a1c7-9e0e-4a19-92fa-f986e8b94cc6.png)

**Reply by balaupp** [reply]
Solved it with the help of this: https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/303
Ad blocker was the issue. I thought I have disabled all (both) of them, but one remained active. Uninstalling did the trick. Thanks!

**Comment by JSekellick** [comment]
Hello everyone. I'm a asset manager trying to learn myself into a new career field of Data Science. I'm late getting started, but hope to learn a lot from everyone and maybe help other where I can.
My project is:
Send a message to a Teams channel or chat

Nice meeting everyone!!

**Reply by aycabas** [reply]
So great to have you hear @Barbuckles, looking forward to see your project up and running! :)

**Comment by pnet2log** [comment]
Hello everyone, my name is Paul Simon Umaru, a Microsoft Learn Student Ambassador and a Power Platform developer from Nigeria.

Hope our learning experience is exciting. Here to offer any assistance.

**Comment by samueljaja00** [comment]
I'm samueljaja00@gmail.coms, I wouldn't finish my program because of my job but I work with Tech  Zain companies which I own with my dichotomies understanding the program of temux  and I willing to work as I'm less busy with no job

**Comment by timothynn** [comment]
> 

Hi, I'm Tim, I am a 3rd year university student from Kenya. I am curious about Microsoft Graph and I hope the learning experience will give me ideas on cool projects. 🤞🏽 ❤️

**Comment by aycabas** [comment]
## This is week 4!🔥

Hi everyone! How is week 4 going? Feel free to comment below 🥳

## This week will be all about working on your own projects! 💻

We provide mentorship opportunities with Microsoft Graph & Advocacy Leaders for the top 3 winning projects, get ready to submit your finalize your project until the end of next week for a chance to win exciting prizes! 💪

ℹ️ If you haven't proposed your project yet, submit your project proposal: [Submit Here](https://aka.ms/30DaysOfMSGraph/Scenarios)

* Catch up with our previous session if you didn't get a chance to join live, [Watch On Demand](https://www.youtube.com/watch?v=5-28a796rr4): Can Microsoft Graph Do What?! Real-World Scenarios for Your Apps.
* Join our ask the experts session on Friday if you have any questions about the challenge or your project: Nov 25 - 3PM GMT Live Reactor Session ([RSVP](https://aka.ms/30DaysOfMSGraph/Reactor/06) | [LIVE](https://www.youtube.com/watch?v=D8OHUhiBvz0))

## Don't Forget!
Please make sure to complete all Microsoft Learn materials provided on Week 1 & Week 2 until **Nov 29**, so you'll be eligible to win the Cloud Skills Challenge prizes! All the learning materials are available to consume on your free time, this is a self-pace learning challenge. If you have questions or need help, please utilize this discussion channel, we are always here to help you 😊🚀

If you haven't registered to the challenge yet, please register, so you can be eligible for the exciting prizes🏆: https://aka.ms/30DaysOfMSGraph/Register

**Comment by balaupp** [comment]
Question: I have two tenants, abc.com and xyz.com. I have a user, user1@abc.com in the first tenant and the same user as a **guest user** in the second tenant. Does the system capture the user's activity (calendar, teams and outlook activity etc) in both tenants? Basically if I wrote my app primarily in the xyz.com tenant and have all the necessary permissions and consents, can we read the user1@abc.com data from xyz.com's graph?

**Reply by aycabas** [reply]
Hi @balaupp, if your app is reading a single tenant only, in your case it's xyz.com, your app can only be used by accounts in xyz tenant. You can build your app for multi tenant use by selecting multi tenant option in AAD app registrations, that makes your app usable by any work or school account. As long as user logins to your app, you can show their calendar, outlook events.

**Reply by balaupp** [reply]
Thanks

**Comment by balaupp** [comment]
Thank you. What about when I want to bulk load data using MGDC and not have a user interacting app? How would that work?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by DonkeyKongJr** [comment]
Hi @japhletnwamu and @aycabas.

Are there any updates on the results of the coding challenge?

Unfortunately we couldn't find any.
